CMakeLists.txt

```diff
@@ -620,7 +620,18 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
         SET(CMAKE_C_LINK_EXECUTABLE "")
 
     elseif(TOOLCHAIN STREQUAL "NONE")
-        ADD_COMPILE_OPTIONS(-fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables -flto -DUSING_LTO  -Wno-maybe-uninitialized -Wno-uninitialized  -Wno-builtin-declaration-mismatch -Wno-nonnull-compare -Werror-implicit-function-declaration)
+        if(DISABLE_LTO STREQUAL "1")
+            SET(FLTO_COMPILE_OPTS "-fno-lto")
+            SET(FLTO_LINK_OPTS "-fno-lto")
+        else()
+            if (DEFINED USING_LTO)
+                SET(FLTO_COMPILE_OPTS "-flto" "-${USING_LTO}")
+            else()
+                SET(FLTO_COMPILE_OPTS "-flto")
+            endif()
+            SET(FLTO_LINK_OPTS "-flto")
+        endif()
+        ADD_COMPILE_OPTIONS(-fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables ${FLTO_COMPILE_OPTS}  -Wno-maybe-uninitialized -Wno-uninitialized  -Wno-builtin-declaration-mismatch -Wno-nonnull-compare -Werror-implicit-function-declaration)
```

Comment:
This is a good improvement.

The [point](https://github.com/DMTF/libspdm/blob/main/doc/build.md#linux-builds-inside-build-environments) of the "NONE" option though was to provide none of these options (and the build project should provide them). It seems like https://github.com/DMTF/libspdm/pull/2464 introduced a bug setting these.

I think it's probably better to revert these changes out to fix the build issues